### ISSUE TITLE
[Report] [4.0] Unable to download the required update.

### ISSUE THREAD

**Issue by ghost** [holoiso-eol/holoiso#689]
## Your Issue:
```
I installed HoloISO and rebooted into it. I select language, timezone and network interface then there is a error came up, it says "Unable to download the required update. Please check your network connection and try again.". I tried even with VPN.
The error shows for a very short time, i couldn't even read it, i recorded a video and took screenshots from it. It goes to shutdown after the error. This was a loop for me.
```

## Your device's full specs
```
CPU: AMD Ryzen 3 4300GE with Radeon Graphics
GPU: AMD ATI 08:00.0 Renoir
Graphics:
  Device-1: AMD Renoir driver: amdgpu v: kernel
  Display: x11 server: X.org v: 1.21.1.6 with: Xwayland v: 22.1.7 driver:
    X: loaded: amdgpu unloaded: modesetting gpu: amdgpu
    resolution: 1366x768~60Hz
  OpenGL: renderer: AMD Radeon Graphics (renoir LLVM 15.0.6 DRM 3.49
  6.1.5-x86_64-xanmod1)
    v: 4.6 Mesa 22.2.5
```

## Your OS version (`cat /etc/os-release | grep VARIANT_ID | cut -d '"' -f 2`):
```
snapshot20221211.1426
```

## Your installation type:
```
Ventoy
```

## Have you updated your installation recently?
```
nawww
```
![2](https://user-images.githubusercontent.com/61632416/212541420-3a482097-b5c5-4ec7-9cd7-1101cc34acc5.jpeg)
![1](https://user-images.githubusercontent.com/61632416/212541422-392a0c76-4201-4c81-bce9-3023b58ab6f0.jpeg)

**Comment by ghost** [comment]
## I tried installing holoiso in a virtual machine. I see those errors, maybe those errors are source of the issue.
![Screenshot_archlinux_2023-01-15_18:47:30](https://user-images.githubusercontent.com/61632416/212551267-2996faa4-9a2d-44c9-af76-1305040e670b.png)
## I chrooted into installed system and then run `pacman -Sy` and it gave errors.
![Screenshot_archlinux_2023-01-15_18:48:27](https://user-images.githubusercontent.com/61632416/212551253-e75b7faa-ba5e-4259-8a4e-f183cb6dd77d.png)

**Comment by aleksandrsma** [comment]
same

**Comment by ghost** [comment]
In the [repository](https://steamdeck-packages.steamos.cloud/archlinux-mirror/), folder names are different than in `/etc/pacman.conf`
For example, jupiter: 
![image](https://user-images.githubusercontent.com/61632416/212568023-1928d041-d80f-422a-b8d9-69cfad0f4014.png)
I have changed repos in `/etc/pacman.conf` with 3.3 ones. For example; [holo-3.3.3] instead of [holo] because there is no [holo]

I could install holoiso with this way without the error.

```
#
# /etc/pacman.conf
#

#[testing-rel]
#Include = /etc/pacman.d/mirrorlist

[holoiso-stable]
Include = /etc/pacman.d/holo_mirrorlist
SigLevel = Never

[holostaging]
Include = /etc/pacman.d/holo_mirrorlist
SigLevel = Never

[jupiter-3.3]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[holo-3.3.3]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[core-3.3]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[extra-3.3]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

#[community-testing-rel]
#Include = /etc/pacman.d/mirrorlist

[community-3.3]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[multilib-3.3]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

```

My full `/etc/pacman.conf` file.
```
#
# /etc/pacman.conf
#
# See the pacman.conf(5) manpage for option and repository directives

#
# GENERAL OPTIONS
#
[options]
# The following paths are commented out with their default values listed.
# If you wish to use different paths, uncomment and update the paths.
#RootDir     = /
#DBPath      = /var/lib/pacman/
#CacheDir    = /var/cache/pacman/pkg/
#LogFile     = /var/log/pacman.log
#GPGDir      = /etc/pacman.d/gnupg/
#HookDir     = /etc/pacman.d/hooks/
HoldPkg     = pacman glibc
#XferCommand = /usr/bin/curl -L -C - -f -o %o %u
#XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
#CleanMethod = KeepInstalled
Architecture = auto

# Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
IgnorePkg = jupiter-hw-support linux-firmware-neptune linux-firmware-neptune-rtw-debug
#IgnoreGroup =

#NoUpgrade   =
#NoExtract   =

# Misc options
#UseSyslog
#Color
#TotalDownload
# We cannot check disk space from within a chroot environment
CheckSpace
#VerbosePkgLists
ParallelDownloads = 5

# By default, pacman accepts packages signed by keys that its local keyring
# trusts (see pacman-key and its man page), as well as unsigned packages.
SigLevel    = Required DatabaseOptional
LocalFileSigLevel = Optional
#RemoteFileSigLevel = Required

# NOTE: You must run `pacman-key --init` before first using pacman; the local
# keyring can then be populated with the keys of all official Arch Linux
# packagers with `pacman-key --populate archlinux`.

#
# REPOSITORIES
#   - can be defined here or included from another file
#   - pacman will search repositories in the order defined here
#   - local/custom mirrors can be added here or in separate files
#   - repositories listed first will take precedence when packages
#     have identical names, regardless of version number
#   - URLs will have $repo replaced by the name of the current repo
#   - URLs will have $arch replaced by the name of the architecture
#
# Repository entries are of the format:
#       [repo-name]
#       Server = ServerName
#       Include = IncludePath
#
# The header [repo-name] is crucial - it must be present and
# uncommented to enable the repo.
#

# The testing repositories are disabled by default. To enable, uncomment the
# repo name header and Include lines. You can add preferred servers immediately
# after the header, and they will be used before the default mirrors.

#[testing-rel]
#Include = /etc/pacman.d/mirrorlist

[holoiso-stable]
Include = /etc/pacman.d/holo_mirrorlist
SigLevel = Never

[holostaging]
Include = /etc/pacman.d/holo_mirrorlist
SigLevel = Never

[jupiter-3.3]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[holo-3.3.3]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[core-3.3]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[extra-3.3]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

#[community-testing-rel]
#Include = /etc/pacman.d/mirrorlist

[community-3.3]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[multilib-3.3]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

# An example of a custom package repository.  See the pacman manpage for
# tips on creating your own repositories.
#[custom]
#SigLevel = Optional TrustAll
#Server = file:///home/custompkgs
```

**Comment by hadamlenz** [comment]
When making the above change I get the following error on `packman -Syu`

```
error: failed to commit transaction (conflicting files)
/usr/bin/jupiter-biosupdate exists in both 'jupiter-hw-support' and 'holoiso-main'
/usr/bin/jupiter-controller-update exists in both 'jupiter-hw-support' and 'holoiso-main'
/usr/bin/steamos-polkit-helpers/steamos-priv-write exists in both 'jupiter-hw-support' and 'holoiso-main'
```

should the suffixes all be `-rel`?

or maybe some other `holoiso` repo

**update**
changing all the suffixes to `-rel` gets further, but same error occurs

**Comment by ghost** [comment]
Which version of the ISO did you install? My `pacman.conf` does not have `jupiter-hw-support` repository.

I have installed 4.0 version of ISO. At first, I have tried with `-rel` suffix but it didnt work, there was same error. Then I tried with `3-3` suffix. I guess suffix `3-3` means old version's repository. You can try installing 4.0 version of ISO and `-main` suffix.

**Comment by hadamlenz** [comment]
@nxjosephofficial the version that is in releases right now, 4.0.  It's a fresh install.

I also dont have a `jupiter-hw-support` in pacman.conf.  it looks like `jupiter-hw-support` is a package [here](https://steamdeck-packages.steamos.cloud/archlinux-mirror/jupiter-main/os/x86_64/)

using the `-main` suffix yielded an extra error

```
/etc/pacman.conf exists in both 'pacman' and 'holoiso-main'
```

[I found someone else with the same issue](https://steamdeck.community/threads/holoiso-main-package-conflicts.352/#post-1910)

**Comment by ghost** [comment]
I have never encountered with this issue may because I didn't wait to the end of `pacman -Syu` command. Does these errors happening at start of `pacman -Syu` command or at end?

Also can you post your `/etc/pacman.conf`

**Comment by hadamlenz** [comment]
Happens at the end


```
#
# /etc/pacman.conf
#
# See the pacman.conf(5) manpage for option and repository directives

#
# GENERAL OPTIONS
#
[options]
# The following paths are commented out with their default values listed.
# If you wish to use different paths, uncomment and update the paths.
#RootDir     = /
#DBPath      = /var/lib/pacman/
#CacheDir    = /var/cache/pacman/pkg/
#LogFile     = /var/log/pacman.log
#GPGDir      = /etc/pacman.d/gnupg/
#HookDir     = /etc/pacman.d/hooks/
HoldPkg     = pacman glibc
#XferCommand = /usr/bin/curl -L -C - -f -o %o %u
#XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
#CleanMethod = KeepInstalled
Architecture = auto

# Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
IgnorePkg = linux-firmware-neptune linux-firmware-neptune-rtw-debug
#IgnoreGroup =

#NoUpgrade   =
#NoExtract   =

# Misc options
#UseSyslog
#Color
#TotalDownload
# We cannot check disk space from within a chroot environment
#CheckSpace
#VerbosePkgLists
ParallelDownloads = 5

# By default, pacman accepts packages signed by keys that its local keyring
# trusts (see pacman-key and its man page), as well as unsigned packages.
SigLevel    = Required DatabaseOptional
LocalFileSigLevel = Optional
#RemoteFileSigLevel = Required

# NOTE: You must run `pacman-key --init` before first using pacman; the local
# keyring can then be populated with the keys of all official Arch Linux
# packagers with `pacman-key --populate archlinux`.

#
# REPOSITORIES
#   - can be defined here or included from another file
#   - pacman will search repositories in the order defined here
#   - local/custom mirrors can be added here or in separate files
#   - repositories listed first will take precedence when packages
#     have identical names, regardless of version number
#   - URLs will have $repo replaced by the name of the current repo
#   - URLs will have $arch replaced by the name of the architecture
#
# Repository entries are of the format:
#       [repo-name]
#       Server = ServerName
#       Include = IncludePath
#
# The header [repo-name] is crucial - it must be present and
# uncommented to enable the repo.
#

# The testing repositories are disabled by default. To enable, uncomment the
# repo name header and Include lines. You can add preferred servers immediately
# after the header, and they will be used before the default mirrors.

#[testing]
#Include = /etc/pacman.d/mirrorlist

[holoiso-stable]
Include = /etc/pacman.d/holo_mirrorlist
SigLevel = Never

[holostaging]
Include = /etc/pacman.d/holo_mirrorlist
SigLevel = Never

[jupiter-main]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[holo-main]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[core-main]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[extra-main]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

#[community-testing]
#Include = /etc/pacman.d/mirrorlist

[community-main]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[multilib-main]
Include = /etc/pacman.d/mirrorlist
SigLevel = Never

# An example of a custom package repository.  See the pacman manpage for
# tips on creating your own repositories.
#[custom]
#SigLevel = Optional TrustAll
#Server = file:///home/custompkgs
```

**Comment by ghost** [comment]
Can you try to overwrite as mentioned in https://github.com/theVakhovskeIsTaken/holoiso/issues/446

`sudo pacman -Syu --overwrite="*"`

**Comment by hadamlenz** [comment]
worked, thank you

**Comment by ghost** [comment]
You are welcome.

**Comment by SteveDevOps** [comment]
So I did the 4.0 iso with ventoy as well and have the bad repo names without any suffix. I'm in the OS playing games successfully -- but can't update -- where it's 404ing and I can see this from the command line. I tried the` --overwrite="*"` switch but that didn't do anything for me -- did you guys add that AFTER changing the repo names to -main ? Is it better to pull down the pacman.conf from the raw repo and overwrite the file first? If so, which?

**Comment by ghost** [comment]
Yes, you should use the `-main` suffix and those errors below are happening at the end of `sudo pacman -Syu` command.
Then you can run `sudo pacman -Syu --overwrite="*"`

**Comment by SteveDevOps** [comment]
I added -main and ran just a plain pacman -Syu .. and  now I'm stuck with what another mentioned above:

`Net Upgrade Size:       398.99 MiB

:: Proceed with installation? [Y/n] y
(840/840) checking keys in keyring                              [##################################] 100%
(840/840) checking package integrity                            [##################################] 100%
(840/840) loading package files                                 [##################################] 100%
(840/840) checking for file conflicts                           [##################################] 100%
error: failed to commit transaction (conflicting files)
/etc/pacman.conf exists in both 'pacman' and 'holoiso-main'
/usr/bin/jupiter-biosupdate exists in both 'jupiter-hw-support' and 'holoiso-main'
/usr/bin/jupiter-controller-update exists in both 'jupiter-hw-support' and 'holoiso-main'
/usr/bin/steamos-polkit-helpers/steamos-priv-write exists in both 'jupiter-hw-support' and 'holoiso-main'
Errors occurred, no packages were upgraded.

**Comment by ghost** [comment]
did you run pacman -Syu without overwrite parameter?

**Comment by SteveDevOps** [comment]
ok now that worked with the `--overwrite="*"` thx.. I think I had done that out of sequence earlier

**Comment by ghost** [comment]
You are welcome.
> where it's 404ing and I can see this from the command line. I tried the --overwrite="*" switch but that didn't do anything for me 

Yes, probably you did it before.

**Comment by SteveDevOps** [comment]
for those that find their way here.. and need the above encapsulated in a one-shot script.. my attempts.. 
https://gist.github.com/SteveDevOps/617a1c9274460b8b055f65eb554cdadb

contributions welcome.

**Comment by SteveDevOps** [comment]
I can also confirm that these system update issues did not occur for me when starting over with the 4.1 nightly built iso -- looks like it was built Jan 9th, after extracting from zip. 
https://nightly.link/theVakhovskeIsTaken/holoiso/workflows/build/stable/holoiso
I was able to update from gui, and didn't have to enter desktop mode to fix things. Appreciate all the hard work from the DEVs.
--I assume this is also due to the rebasing commit that occurred off SteamOS 3.4.

**Comment by khvolk** [comment]
I did not succeed to solve my upgrade problems at all, untill I used:

https://github.com/theVakhovskeIsTaken/holoiso/blob/stable/pacman.conf

Now everything works.

**Comment by KingOfFalcons** [comment]
> ## Your Issue:
> ```
> I installed HoloISO and rebooted into it. I select language, timezone and network interface then there is a error came up, it says "Unable to download the required update. Please check your network connection and try again.". I tried even with VPN.
> The error shows for a very short time, i couldn't even read it, i recorded a video and took screenshots from it. It goes to shutdown after the error. This was a loop for me.
> ```
> 
> ## Your device's full specs
> ```
> CPU: AMD Ryzen 3 4300GE with Radeon Graphics
> GPU: AMD ATI 08:00.0 Renoir
> Graphics:
>   Device-1: AMD Renoir driver: amdgpu v: kernel
>   Display: x11 server: X.org v: 1.21.1.6 with: Xwayland v: 22.1.7 driver:
>     X: loaded: amdgpu unloaded: modesetting gpu: amdgpu
>     resolution: 1366x768~60Hz
>   OpenGL: renderer: AMD Radeon Graphics (renoir LLVM 15.0.6 DRM 3.49
>   6.1.5-x86_64-xanmod1)
>     v: 4.6 Mesa 22.2.5
> ```
> 
> ## Your OS version (`cat /etc/os-release | grep VARIANT_ID | cut -d '"' -f 2`):
> ```
> snapshot20221211.1426
> ```
> 
> ## Your installation type:
> ```
> Ventoy
> ```
> 
> ## Have you updated your installation recently?
> ```
> nawww
> ```
> 
> ![2](https://user-images.githubusercontent.com/61632416/212541420-3a482097-b5c5-4ec7-9cd7-1101cc34acc5.jpeg) ![1](https://user-images.githubusercontent.com/61632416/212541422-392a0c76-4201-4c81-bce9-3023b58ab6f0.jpeg)

i have exactly the same issue on a Minisforum UM 690. Logged a report as well before i saw this. Anyone figure out a way to solve it?

**Comment by KingOfFalcons** [comment]
> I did not succeed to solve my upgrade problems at all, untill I used:
> 
> https://github.com/theVakhovskeIsTaken/holoiso/blob/stable/pacman.conf
> 
> Now everything works.

i'm really sorry, but i'm new to SteamOS and linux. Can you please explain how i can replace the pacman.conf file with the version you just posted? i know how to run command lines in general, but just not in linux / Steam OS / Holo installer environment.

**Comment by KingOfFalcons** [comment]
ji have 

> I can also confirm that these system update issues did not occur for me when starting over with the 4.1 nightly built iso -- looks like it was built Jan 9th, after extracting from zip. https://nightly.link/theVakhovskeIsTaken/holoiso/workflows/build/stable/holoiso I was able to update from gui, and didn't have to enter desktop mode to fix things. Appreciate all the hard work from the DEVs. --I assume this is also due to the rebasing commit that occurred off SteamOS 3.4.

i can confirm that with this image everything was resolved for me as well.

**Comment by khvolk** [comment]
You edit /etc/pacman.conf. Delete all Lines. Open 

https://github.com/theVakhovskeIsTaken/holoiso/blob/stable/pacman.conf

in a Webbrowser. Click on the "raw" Tab. <CTRL> and A will select the whole text. Copy and Paste this to /etc/pacman.conf. Save it.

Run   pacman -Syu   as root user to update all.

**Comment by KingOfFalcons** [comment]
@khvolk  i managed to resolve the issue by using the nightly build from early january. but for future learning i would love to understand how to do this during the installation process. In my case have no internet connectivity when loading the installer usb and only get internet connectivity once SteamOS is intalled, but at that point there doesnt seem to be an ability to access a command prompt to do the editing you describe....

or was the solution only meant for people who already have succeeded in installing SteamOS and can update those files in Desktop mode?

PS. thanks for all the great work you are doing on this. it is really great to see SteamOS become accessible to a wider hardware set like this. much appreciated!!!

**Comment by khvolk** [comment]
@KingOfFalcons what we are doing here with the SteamOS is installing an Archlinux with a special Steam-Client which is made for the Steam-Deck. If you want to learn more on and play with Linux, you could take a Desktop-Computer (with AMD-CPU like 3200G or 5600G) install Archlinux on it and a Steam client. More convenient is perhaps using a distribution like EndeavourOS. This is Archlinux with a graphical installer (in short).

With an ArchLinux Install-USB-Stick you can mount Linux- (or Windows-) Partitions for maintenance at any time and modify or backup files.

**Comment by KingOfFalcons** [comment]
sorry to start this up again. i got a new harddisk and just tried installing the version i got throught the nightly link. i managed to get through the install and login to steam, but am now stuck in an infinite update loop. it says it is extracting package, installing update, verifying update, and then restarts again to do the same thing. is this part of the same problem you think? that can be resolved in the way described above...? i dont even know how to create an arch linux usb and/or mount a drive in it... so this will be interesting.

**Comment by SteveDevOps** [comment]
Curious.. is the new drive a regular SSD 2.5, or an nvme?

**Comment by KingOfFalcons** [comment]
2.5 ssd. I was running off usb before. Terribly slow. Way quicker now. Why?

**Comment by KingOfFalcons** [comment]
I went back to an earlier release from november and am in... But worried
about attempting update to steam os

**Comment by khvolk** [comment]
KingOfFalcons. Do you want me to explain how to copy your running installation from the USB-stick to the SSD using rsync?

**Comment by MickWild** [comment]
this config work for me
```
#
# /etc/pacman.conf
#
# See the pacman.conf(5) manpage for option and repository directives

#
# GENERAL OPTIONS
#
[options]
# The following paths are commented out with their default values listed.
# If you wish to use different paths, uncomment and update the paths.
#RootDir     = /
#DBPath      = /var/lib/pacman/
#CacheDir    = /var/cache/pacman/pkg/
#LogFile     = /var/log/pacman.log
#GPGDir      = /etc/pacman.d/gnupg/
#HookDir     = /etc/pacman.d/hooks/
HoldPkg     = pacman glibc
#XferCommand = /usr/bin/curl -L -C - -f -o %o %u
#XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
#CleanMethod = KeepInstalled
Architecture = auto

# Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
IgnorePkg = linux-firmware-neptune linux-firmware-neptune-rtw-debug
#IgnoreGroup =

#NoUpgrade   =
#NoExtract   =

# Misc options
#UseSyslog
#Color
#TotalDownload
# We cannot check disk space from within a chroot environment
#CheckSpace
#VerbosePkgLists
ParallelDownloads = 5

# By default, pacman accepts packages signed by keys that its local keyring
# trusts (see pacman-key and its man page), as well as unsigned packages.
SigLevel    = Required DatabaseOptional
LocalFileSigLevel = Optional
#RemoteFileSigLevel = Required

# NOTE: You must run `pacman-key --init` before first using pacman; the local
# keyring can then be populated with the keys of all official Arch Linux
# packagers with `pacman-key --populate archlinux`.

#
# REPOSITORIES
#   - can be defined here or included from another file
#   - pacman will search repositories in the order defined here
#   - local/custom mirrors can be added here or in separate files
#   - repositories listed first will take precedence when packages
#     have identical names, regardless of version number
#   - URLs will have $repo replaced by the name of the current repo
#   - URLs will have $arch replaced by the name of the architecture
#
# Repository entries are of the format:
#       [repo-name]
#       Server = ServerName
#       Include = IncludePath
#
# The header [repo-name] is crucial - it must be present and
# uncommented to enable the repo.
#

# The testing repositories are disabled by default. To enable, uncomment the
# repo name header and Include lines. You can add preferred servers immediately
# after the header, and they will be used before the default mirrors.

#[testing]
#Include = /etc/pacman.d/mirrorlist

[holoiso-stable]
Include = /etc/pacman.d/holo_mirrorlist
SigLevel = Never

[holostaging]
Include = /etc/pacman.d/holo_mirrorlist
SigLevel = Never

[jupiter-rel]
Server = https://steamdeck-packages.steamos.cloud/archlinux-mirror/$repo/os/$arch
#Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[holo-rel]
Server = https://steamdeck-packages.steamos.cloud/archlinux-mirror/$repo/os/$arch
#Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[core-rel]
Server = https://steamdeck-packages.steamos.cloud/archlinux-mirror/$repo/os/$arch
#Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[extra-rel]
Server = https://steamdeck-packages.steamos.cloud/archlinux-mirror/$repo/os/$arch
#Include = /etc/pacman.d/mirrorlist
SigLevel = Never

#[community-testing]
#Include = /etc/pacman.d/mirrorlist

[community-rel]
Server = https://steamdeck-packages.steamos.cloud/archlinux-mirror/$repo/os/$arch
#Include = /etc/pacman.d/mirrorlist
SigLevel = Never

[multilib-rel]
Server = https://steamdeck-packages.steamos.cloud/archlinux-mirror/$repo/os/$arch
#Include = /etc/pacman.d/mirrorlist
SigLevel = Never

# An example of a custom package repository.  See the pacman manpage for
# tips on creating your own repositories.
#[custom]
#SigLevel = Optional TrustAll
#Server = file:///home/custompkgs
```
I play in Intel Hades Canyon NUC

**Comment by ghost** [comment]
> i got a new harddisk and just tried installing the version i got throught the nightly link.
am now stuck in an infinite update loop. it says it is extracting package, installing update, verifying update, and then restarts again to do the same thing.

Try installing holoiso `4.0`. 

I guess nightly build just use `-rel` suffix on `pacman.conf`. You can add `-rel` suffix by chrooting into installed holoiso system after installing it with holoiso installation script that located on the desktop. Chrooting script is located on the desktop too.

**Comment by KingOfFalcons** [comment]
> KingOfFalcons. Do you want me to explain how to copy your running installation from the USB-stick to the SSD using rsync?

@khvolk  that is very generous, but i've already wiped it and used your november image to install Steam OS. Games are working mostly fine, but it wont update which gives an error. I'm playing around with the desktop mode and figuring out if i can update the pacman.conf file there... otherwise, i'm happy to wait for your next official release where i'm sure this will be resolved... 

appreciate all the effort you do on this.

**Comment by KingOfFalcons** [comment]
@khvolk  so i managed to find and edit the pacman.conf file in desktop mode. modifying the repositories with "-rel" as shown above seems to have resolved the system update issue. yay! learning as i go along :-)

at this point, is there any reason why i should want to start over and re-install everything with v4.0? or will the update have taken care of things? sorry to bother you with all these questions. appreciate your help!

PS. the last thing i want to change is the default selection on boot loader. this is now neptune, but that doesnt load my ethernet connection. so prefer to load the holoiso image. havent figured out how to do that yet. i tried download grub configurator but that wont load. not sure what to try next as everything i found online points at grub configurator.

**Comment by RaidOpe** [comment]
After my tests, HOLO 4.0 didn't work on the steam deck, it stuck in a reboot loop after selecting wifi, which made me very annoyed, if I could go into the system to update manually, I would be able to use this system. This is a vicious bug

**Comment by KingOfFalcons** [comment]
Raid. The work around was shared a few messages ago. One if two things
should work.

1. You download the image from the nightly link i shared above. It worked
for me once but not the second time. Try it. If it works it is the easiest
fix

2. If that doesn't work you not again from the instructions usb. Ince the
desktop loads you can use the chroot link on the top left to access your
installation drive. In there you need to update the address where steamos
gets it's updates. Above there is a print of the file you need to update
and what it needs to look like. You essentially need to add -rel to a bunch
of lines at the bottom of the .conf file. It's not hard, if you know how to
find and edit file using the chroot command prompt. I don't do i used v3.4
to do the install. If you do know please share with me

One if those will definitely work for you

Remember. This is a hobby project. Nobody is making money from this. And
people have real lives. Let's just be grateful that this exists and not
complain to much. I agree it's an annoying bug and it cost me 3 days to
figure it out. But that is what happens when you trying new things like
this. Part of the fun i guess. Opportunity to learn new things

Good luck. And us know how you get on

**Comment by AlghamdiAbdul** [comment]
i just installed holoiso and it gives me the same message which is "failed to update required files"  i tried editing /etc/pacman.conf but it won't let me save it. it gives me a pop-up where it tells me to put the root password when ever i do it tells me authentication failed. any help on that?

**Comment by SteveDevOps** [comment]
When you ran the install initially, you had to specify root pw... it's that
pw... apparently you're fat fingering it / getting it wrong -- hence the
error. Or when you initially specified it.. you didn't type in what you
thought you did due to mistype or capslock, etc.

**Comment by AlghamdiAbdul** [comment]
i tried putting the password that i initially put when the installer asked me to put a root password but it doesn't work for some reason. also i tried reinstalling and putting another password that i was certain that it was right but still doesn't work for me

**Comment by SteveDevOps** [comment]
Try getting access to /etc/pacman.conf by using your user account's password instead..

Try this from terminal.. and you'll probably get the same result.. but type `sudo su -` (which is become root by using sudo privs) .. and type in your **user account pw** which has sudo rights/privs. Then you can go edit the file as root.

or you can type`sudo vim /etc/pacman.conf`  to go edit the file as root directly, by using sudo privs. If you're not familiar with vim shortcuts for editing, then do `sudo nano /etc/pacman.conf`

furthermore, if that all works... and you're able to gain root access through sudo -- then you can fix root's pw from the cmd line as well and type `passwd` after `sudo su -`  and reset root's pw.

**Comment by Burgersim** [comment]
I have switched the -rel in pacman.conf but now when I try to run sudo pacman -Syu I get a lot of corrupted packages and wrong signature error and it's asking me to delete or keep the files but neither results in any upgrade or update when I try again afterwards, just always the same error.

-rel or -main did not make any difference for me btw. I also tried to init and generate pacman keyring again, to no avail

**Comment by SteveDevOps** [comment]
@Burgersim I would start over at this point.. grab the 4.1 nightly build and reinstall: https://nightly.link/theVakhovskeIsTaken/holoiso/workflows/build/stable/holoiso

**Comment by crashedash32** [comment]
Hi, i am trying to update HoloISO OS, i have done a fresh install today, my hardware is:

AMD V1605b SoC
32GB RAM
1TB 2.5" SSD

it runs through the initial steam setups, installs all updates in the steam OS part, latest update was 29th of March.

I then drop to desktop and would like to update this, so i get latest glibc etc..

i have tried adding -rel to pacman.conf and -main, i have tried --overwrite='*' 

but i get the following error: 
error: extract: not overwriting dir with file /boot/efi
error: could not commit transaction
error: failed to commit transaction (transaction aborted)
Errors occurred no packages were upgraded

i orginally had the same issues as above with a duplicate package in jupiter-hw..

i have hosed the OS twice.. LOL..

i'm sure i am missing the obvious, but i really dont know what that is..

thank for the fantastic work you have done here and thanks in advance for any advice offered.. :)

**Comment by crashedash32** [comment]
> Hi, i am trying to update HoloISO OS, i have done a fresh install today, my hardware is:
> 
> AMD V1605b SoC 32GB RAM 1TB 2.5" SSD
> 
> it runs through the initial steam setups, installs all updates in the steam OS part, latest update was 29th of March.
> 
> I then drop to desktop and would like to update this, so i get latest glibc etc..
> 
> i have tried adding -rel to pacman.conf and -main, i have tried --overwrite='*'
> 
> but i get the following error: error: extract: not overwriting dir with file /boot/efi error: could not commit transaction error: failed to commit transaction (transaction aborted) Errors occurred no packages were upgraded
> 
> i orginally had the same issues as above with a duplicate package in jupiter-hw..
> 
> i have hosed the OS twice.. LOL..
> 
> i'm sure i am missing the obvious, but i really dont know what that is..
> 
> thank for the fantastic work you have done here and thanks in advance for any advice offered.. :)

Strange update.. 

i ran exactly the same command again

sudo pacman -Syu --overwrite='*' 

and this time it worked?

however, i now have no mouse gaming mode, is that expected with latest updates installed?

**Comment by jasjnr** [comment]
Any ideas why the links to the builds are 404? Trying to update using 4.1 as 4.0 is not booting on an Intel NUC

**Comment by n3rdw1z4rd** [comment]
> @Burgersim I would start over at this point.. grab the 4.1 nightly build and reinstall: https://nightly.link/theVakhovskeIsTaken/holoiso/workflows/build/stable/holoiso

@SteveDevOps : When I click on the link you provided, I get the following result:
![image](https://github.com/HoloISO/holoiso/assets/33690133/03604eea-e606-4e8c-9d3b-a70d99ee556f)

In fact, for any of the links to nightly builds, I get a 404. I would like to attempt to build locally, can you provide the commit sig for the 4.1 nightly build you linked so I can build the same one please?

Thanks :)

**Comment by SteveDevOps** [comment]
@JohnCWakley it appears their nightly page is down. I highly recommend ChimeraOS i have moved on to that distro.

**Comment by n3rdw1z4rd** [comment]
> @JohnCWakley it appears their nightly page is down. I highly recommend ChimeraOS i have moved on to that distro.

Thanks for the reply @SteveDevOps, but isn't ChimeraOS still based on Debian and SteamOS 3?

**Comment by CasperTheEpic** [comment]
> > @JohnCWakley it appears their nightly page is down. I highly recommend ChimeraOS i have moved on to that distro.
> 
> Thanks for the reply @SteveDevOps, but isn't ChimeraOS still based on Debian and SteamOS 3?


Its using Gnome desktop, but its built around the same concept of HaloISO.

I wish a new version of HoloISO would come out with the correct conf file cause even following this it breaks my install 🙃 it would almost be better to just use Manjaro in bigpicture mode until Valve releases an official iOS

**Comment by n3rdw1z4rd** [comment]
> @CasperTheEpic  I wish a new version of HoloISO would come out with the correct conf file cause even following this it breaks my install 🙃 it would almost be better to just use Manjaro in bigpicture mode until Valve releases an official iOS

I built the ISO from the instructions [HERE](https://github.com/HoloISO/holoiso#building-the-iso). You will need a PC with ArchLinux in order to do this correctly. I do have the ISO I built (which does work), but I don't have any means of securely sharing it with a large amount of people.

If anyone has a free file sharing service recommendation, let me know. The ISO is 2.61 GB, so the service will need to allow for that, free.
